### PR TITLE
docs(readme): best practices for imports

### PR DESCRIPTION
**Pull request by polomani**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -95,6 +95,39 @@ Samples are in the [samples/](./samples) directory. The samples' [README.md](./s
 - [API documentation](https://doc.cognitedata.com)
 - [API reference documentation](https://doc.cognitedata.com/api/v1)
 
+## Best practices
+
+### No submodule imports
+
+We highly recommend avoid importing anything from internal sdk modules.
```
Comment (hhenriques) [inline README.md]:
```suggestion
We highly recommend avoiding importing anything from internal SDK modules.
```

File: README.md [hunk 2]
```diff
@@ -95,6 +95,39 @@ Samples are in the [samples/](./samples) directory. The samples' [README.md](./s
...
+```
+
+**Good:**
+```
+import { CogniteAsyncIterator } from '@cognite/sdk'; // ✅
+
+let assetsApi: CogniteClient['assets']; // ✅
+```
+
+You may leverage [eslint](https://eslint.org/docs/rules/no-restricted-imports) to free yourself from memorizing those rules:
```
Comment (hhenriques) [inline README.md]:
```suggestion
We recommend the usage of [eslint](https://eslint.org/docs/rules/no-restricted-imports) to ensure this best practice is enforced without having to memorize the patterns:
```

**Review by hhenriques — CHANGES_REQUESTED** [review]
A great addition to the documentation. Just suggest some minor changes to the wording

**Comment by polomani** [comment]
We might need to rebase this on v3 branch

**Comment by polomani** [comment]
A added this to v3 directly

**Pull request closed without merging**